test/multiplexed-session.ts

```diff
@@ -0,0 +1,326 @@
+      assert.strictEqual(_refreshStub.callCount, 1);
+    });
+  });
+
+  describe('_createSession', () => {
+    it('should create the mux sessions', async () => {
+      await multiplexedSession._createSession();
+      assert.strictEqual(createSessionStub.callCount, 1);
+    });
+
+    it('should pass the multiplexed option', async () => {
+      const multiplexed = {multiplexed: true};
+      await multiplexedSession._createSession();
+      assert.deepStrictEqual(createSessionStub.lastCall.args[0], multiplexed);
+    });
+
+    it('should not make multiple requests', async () => {
+      createSessionStub.onCall(0).resolves([[{}, {}]]);
+      createSessionStub.onCall(1).resolves([[{}]]);
+
+      await multiplexedSession._createSession();
+
+      assert.strictEqual(createSessionStub.callCount, 1);
+    });
+
+    it('should reject with any request errors', async () => {
+      const error = new Error('err');
+      createSessionStub.rejects(error);
+
+      try {
+        await multiplexedSession._createSession();
+        throw new Error('Should not make it this far.');
+      } catch (e) {
+        assert.strictEqual(e, error);
+      }
+    });
+  });
+
+  describe('getSession', () => {
+    it('should acquire a session', done => {
+      sandbox.stub(multiplexedSession, '_acquire').resolves(fakeMuxSession);
+
+      multiplexedSession.getSession((err, session) => {
+        assert.ifError(err);
+        assert.strictEqual(session, fakeMuxSession);
+        done();
+      });
+    });
+    it('should pass any errors to the callback', done => {
+      const error = new Error('err');
+      sandbox.stub(multiplexedSession, '_acquire').rejects(error);
+      multiplexedSession.getSession(err => {
+        assert.strictEqual(err, error);
+        done();
+      });
+    });
+    it('should pass back the session and txn', done => {
+      const fakeTxn = new FakeTransaction() as unknown as Transaction;
+
+      fakeMuxSession.txn = fakeTxn;
+
+      sandbox.stub(multiplexedSession, '_acquire').resolves(fakeMuxSession);
+
+      multiplexedSession.getSession((err, session, txn) => {
+        assert.ifError(err);
+        assert.strictEqual(session, fakeMuxSession);
+        assert.strictEqual(txn, fakeTxn);
+        done();
+      });
+    });
+  });
+
+  describe('_acquire', () => {
+    it('should return a session', async () => {
+      sandbox.stub(multiplexedSession, '_getSession').resolves(fakeMuxSession);
+      const session = await multiplexedSession._acquire();
+      assert.strictEqual(session, fakeMuxSession);
+    });
+
+    it('should have the multiplexed property set to true', async () => {
+      sandbox.stub(multiplexedSession, '_getSession').resolves(fakeMuxSession);
+      const session = await multiplexedSession._acquire();
+      assert.strictEqual((await session!.getMetadata()).multiplexed, true);
+      assert.strictEqual(
+        (await fakeMuxSession!.getMetadata()).multiplexed,
+        true
+      );
+    });
```

Comment:
I feel this test case is not required, as we are just verifying the `getMetadata` stub here, nothing else.

Reply:
certainly, but we are also verifying if the session is containing the multiplexed field enabled to true